src/main/groovy/org/shipkit/internal/gradle/VersionUpgradeConsumerPlugin.java

```diff
@@ -9,170 +9,164 @@
 public class VersionUpgradeConsumerPlugin implements Plugin<Project> {
 
     private static final Logger LOG = Logging.getLogger(VersionUpgradeConsumerPlugin.class);
 
-    public static final String VERSION_UPGRADE_CHECKOUT_BASE_BRANCH = "versionUpgradeCheckoutBaseBranch";
-    public static final String VERSION_UPGRADE_CHECKOUT_VERSION_BRANCH = "versionUpgradeCheckoutVersionBranch";
-    public static final String VERSION_UPGRADE_REPLACE_VERSION = "versionUpgradeReplaceVersion";
-    public static final String VERSION_UPGRADE_GIT_COMMIT = "versionUpgradeGitCommit";
-    public static final String VERSION_UPGRADE_GIT_PUSH = "versionUpgradeGitPush";
-    public static final String VERSION_UPGRADE_CREATE_PULL_REQUEST = "versionUpgradeCreatePullRequest";
+    public static final String CHECKOUT_BASE_BRANCH = "checkoutBaseBranch";
+    public static final String CHECKOUT_VERSION_BRANCH = "checkoutVersionBranch";
+    public static final String REPLACE_VERSION = "replaceVersion";
+    public static final String COMMIT_VERSION_UPGRADE = "commitVersionUpgrade";
+    public static final String PUSH_VERSION_UPGRADE = "pushVersionUpgrade";
+    public static final String CREATE_PULL_REQUEST = "createPullRequest";
     public static final String PERFORM_VERSION_UPGRADE = "performVersionUpgrade";
 
     public static final String DEPENDENCY_NEW_VERSION = "dependencyNewVersion";
-    public static final String DEPENDENCY_BUILD_FILE = "dependencyBuildFile";
-    public static final String DEPENDENCY_PATTERN = "dependencyPattern";
-    public static final String BASE_BRANCH = "baseBranch";
-
-    public static final String DEPENDENCY_PATTERN_DEFAULT = "org.shipkit:shipkit:{VERSION}";
-    public static final String BUILD_FILE_DEFAULT = "build.gradle";
-    public static final String BASE_BRANCH_DEFAULT = "master";
-
-
-    private EnvVariables envVariables;
 
-    @Inject
-    public VersionUpgradeConsumerPlugin(){
-        this(new EnvVariables());
-    }
-
-    @ExposedForTesting
-    public VersionUpgradeConsumerPlugin(EnvVariables envVariables){
-        this.envVariables = envVariables;
-    }
+    private VersionUpgrade versionUpgrade;
 
     @Override
     public void apply(final Project project) {
         LOG.lifecycle("Applying VersionUpgradeConsumerPlugin, beware that it's is INCUBATING state, so its API may change!");
         final ReleaseConfiguration conf = project.getPlugins().apply(ReleaseConfigurationPlugin.class).getConfiguration();
 
-        TaskMaker.task(project, VERSION_UPGRADE_CHECKOUT_BASE_BRANCH, GitCheckOutTask.class, new Action<GitCheckOutTask>() {
+        versionUpgrade = project.getExtensions().create("versionUpgrade", VersionUpgrade.class);
+
+        // set defaults
+        versionUpgrade.setBuildFile(project.file("build.gradle"));
+        versionUpgrade.setBaseBranch("master");
+
+        String dependencyNewVersion = (String) project.getProperties().get(DEPENDENCY_NEW_VERSION);
+        if(dependencyNewVersion != null) {
```

Comment:
I suggest to push out all that complexity elsewhere, for ease of testing. Perhaps to the DependencyNewVersionParser?